src/main/scala/org/constellation/domain/transaction/PendingTransactionsMemPool.scala

```diff
@@ -15,10 +15,20 @@ class PendingTransactionsMemPool[F[_]: Concurrent](semaphore: Semaphore[F])
       if (txs.size < 1) {
         (txs, none[List[TransactionCacheData]])
       } else {
-        val sorted = txs.toList.sortWith(_._2.transaction.edge.data.fee > _._2.transaction.edge.data.fee)
+        val sorted = sortForPull(txs.toList.map(_._2))
         val (left, right) = sorted.splitAt(maxCount)
-        (right.toMap, left.map(_._2).some)
+        (right.map(tx => tx.hash -> tx).toMap, left.some)
       }
     }
 
+  private def sortForPull(txs: List[TransactionCacheData]): List[TransactionCacheData] =
+    txs
+      .groupBy(_.transaction.src.address)
+      .mapValues(_.sortBy(_.transaction.lastTxRef.ordinal))
+      .toSeq
```

Comment:
```scala
.values
.toSeq
.sortBy(_.map(-_.transaction.fee.getOrElse(0L)).sum)
```

If you do not care about the keys then just use `.values` in the very beginning and thanks to that you can get rid ot ugly `._2` and late mapping to values.